microprofile/WFLY-15405_microprofile_reactive_messaging_amqp.adoc

```diff
@@ -0,0 +1,126 @@
+* It must be possible to send and receive messages from an AMQP message broker by using the standard MicroProfile Reactive Messaging constructs (@Incoming/@Outgoing annotated methods, as well as @Channel annotated Emitter and Publisher)
+* Version 1.0 of the AMQP protocol will be supported
+* MicroProfile Config using the properties documented by the https://smallrye.io/smallrye-reactive-messaging/3.16.0/amqp/amqp/[SmallRye Reactive Messaging AMQP Connector] (note the docs are spread over a few pages) are used to map the MicroProfile Reactive Messaging channels to an AMQP broker
+* If an SSLContext is needed (for example to configure truststores to connect to a secured broker which does not have a signed certificate), that is configured in a MicroProfile Config property referencing an SSLContext set up in the Elytron subsystem.
+** The format of the property is `<prefix>.wildfly.elytron.ssl.context=<Elytron SSLContext name>`. For example, to have an `@Outgoing` annotated method sending to the `source` channel secured via the `amqp-ssl-test` SSLContext from Elytron the property entry would look like `mp.messaging.outgoing.source.wildfly.elytron.ssl.context=kafka-ssl-test`.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+* Other AMQP protocol versions than 1.0 will not be supported at this time
+* SmallRye Reactive Messaging's AMQP Connector has a https://smallrye.io/smallrye-reactive-messaging/3.16.0/amqp/client-customization/[CDI mechanism] to do additional configuration of secured connections. This will not be supported. After discussions with the SmallRye team, it seems this mechanism was added in order to have something in this area, but the main polish work has been for their Kafka connector.
+** Although we won't support this, no steps will be taken to stop people from using this mechanism apart from the fact that the classes needed to implement this mechanism are in private modules. Also, we will not, at present, expose the involved jars in our BOMs
+*** Whether this mechanism works or not is unknown, untested and in my opinion should not be documented
+** Rather than using this mechanism we should encourage users to use an SSLContext from the Elytron subsystem
+
+
+== Backwards Compatibility
+
+No backwards compatibility issues. This is an addition to the existing functionality, and does not replace or modify what is already there in any way. The feature as a whole, assuming that the MicroProfile Reactive Messaging subsystem is enabled, is driven by the presence of the modules provided by the `microprofile-reactive-messaging-amqp` Galleon layer.
+
+//=== Default Configuration
+//
+//=== Importing Existing Configuration
+//
+//=== Deployments
+//
+//=== Interoperability
+
+
+== Security Considerations
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+I don't see any security implications.
```

Comment:
I think this sentence could be removed. :) Since this RFE relies on an Elytron `SSLContext`, it does relate to security. The rest of the analysis looks good to me though.